### PR TITLE
Fix worker error silent

### PR DESCRIPTION
**Pull request by hougangliu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/863)

### CODE REVIEW

File: cmd/metricscollector/v1alpha3/file-metricscollector/main.go [hunk 1]
```diff
@@ -58,10 +62,40 @@ var pollInterval = flag.Duration("p", common.DefaultPollInterval, "Poll interval
...
+	for {
+		if line, err := reader.ReadString('\n'); err != nil || io.EOF == err {
+			break
+		} else {
+			klog.Info(line)
+		}
+	}
+	cmd.Wait()
+}
+
```
Comment (gaocegege) [inline cmd/metricscollector/v1alpha3/file-metricscollector/main.go]: Should we use the third-party tool such as https://github.com/hpcloud/tail?

Comment (hougangliu) [reply cmd/metricscollector/v1alpha3/file-metricscollector/main.go]: cool, I can update it

**Comment by johnugeorge** [comment]
Shall we add a failure e2e test to test failed cases?

File: cmd/metricscollector/v1alpha3/file-metricscollector/main.go (hunk 1 above)
Comment (johnugeorge) [reply cmd/metricscollector/v1alpha3/file-metricscollector/main.go]: How do you signal the completion of tail logs?

Comment (hougangliu) [reply cmd/metricscollector/v1alpha3/file-metricscollector/main.go]: we doesn't need know the completion of tail logs, when the worker container exit, metrics collector main process will start to collect and parse the file, and then exit.

**Comment by hougangliu** [comment]
> Shall we add a failure e2e test to test failed cases?

Sure, I can try to add one in another PR

File: cmd/metricscollector/v1alpha3/file-metricscollector/main.go (hunk 1 above)
Comment (johnugeorge) [reply cmd/metricscollector/v1alpha3/file-metricscollector/main.go]: Just wondering if there is a chance to miss certain worker container logs in this case.

**Comment by johnugeorge** [comment]
Please rebase when #861 is merged

File: pkg/webhook/v1alpha3/pod/inject_webhook.go [hunk 2]
```diff
@@ -244,7 +244,7 @@ func wrapWorkerContainer(pod *v1.Pod, jobKind, metricsFile string, mc common.Met
 		if c.Args != nil {
 			args = append(args, c.Args...)
 		}
-		redirectStr := fmt.Sprintf(" 2>&1 | tee %s", metricsFile)
+		redirectStr := fmt.Sprintf("1>%s 2>&1", metricsFile)
```
Comment (johnugeorge) [inline pkg/webhook/v1alpha3/pod/inject_webhook.go]: How do we let users know that `kubectl logs` works on the side car which outputs logs from worker container too?  Should we rename the side car container to "logger" or something?
WDYT? @gaocegege @hougangliu

Comment (gaocegege) [reply pkg/webhook/v1alpha3/pod/inject_webhook.go]: SGTM

Comment (hougangliu) [reply pkg/webhook/v1alpha3/pod/inject_webhook.go]: SGTM

**Comment by gaocegege** [comment]
Needs a reabse.

**Comment by hougangliu** [comment]
> Needs a reabse.

rebased already. It seems CI starts to strike now again

**Comment by hougangliu** [comment]
@gaocegege @johnugeorge 
test passed after nearly 50m, katib CI works slowly now again

**Comment by gaocegege** [comment]
I think we already set it to 3 in test 

https://github.com/kubeflow/katib/blob/master/test/e2e/v1alpha3/run-e2e-experiment.go#L61

**Comment by hougangliu** [comment]
> I think we already set it to 3 in test
> 
> https://github.com/kubeflow/katib/blob/master/test/e2e/v1alpha3/run-e2e-experiment.go#L61

ok

**Pull request merged**